src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -42,7 +42,10 @@ public class DocumentConventions : Client.Conventions
         internal static readonly DocumentConventions DefaultForServer = new DocumentConventions
         {
             SendApplicationIdentifier = false,
-            MaxContextSizeToKeep = new Size(PlatformDetails.Is32Bits == false ? 8 : 2, SizeUnit.Megabytes)
+            MaxContextSizeToKeep = new Size(PlatformDetails.Is32Bits == false ? 8 : 2, SizeUnit.Megabytes),
+#if NETCOREAPP3_1_OR_GREATER
+            HttpPooledConnectionLifetime = TimeSpan.FromMinutes(3)
```

Comment:
Do we have to worry about performance issues here? If I understand properly, it means that every 3 minutes, we'll need to establish a new connection, no?

Reply:
This is for server-side and this is transparently handled by the framework. If connection is used, it is not being disconnected and reconnected. There might be some super small overhead, but server mostly gets TCP info via executor and switched directly to TCP, which is not affected.

Reply:
One place where this happens a _lot_ is the `forward to leader` scenario. We saw some cases where we have > 1K/sec requests in this scenario.

And the issue is that your connection lifetime is now 3 minutes. Assume that establishing a connection take a while (common in distributed scenarios), and now compare exchange or cluster wide transaction will suffer latency spike every 3 minutes, no?

Reply:
Yet, it is mechanism created by MS to protect for DNS and connectivity issues. I can turn it off, but then cluster is getting stuck permanently.